### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by kepta**
Adds pagination to the `task/item` endpoint.
- the default page size 100.
- It is a base 0 indexing.
- Using `page=X` and `page_size=Y` as the query syntax

cc @batpad @mcwhittemore

### CODE REVIEW

**Comment by batpad** [comment]
This looks good to me @kepta - I do think we will need pagination right away to start working on the frontend, so I'm 👍 on merging once @mcwhittemore has a chance to review.

File: lib/helper/sanitizeNumber.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * parses a non number value and returns the default
+ * in case NaN.
+ * @param {*} num 
+ * @param {number} defaultVal 
+ */
+module.exports = function sanitizeNumber(num, defaultVal) {
+  const parsed = parseInt(num, 10);
+  return Number.isNaN(parsed) ? defaultVal : parsed;
```
Comment (mcwhittemore) [inline lib/helper/sanitizeNumber.js]: Should be greater than 0 as well. You handle 0 here but not negative numbers.

File: lib/server.js [hunk 2]
```diff
@@ -150,12 +151,14 @@ server.get('/tasks/:task/stats', function(req, res, next) {
...
 /**
- * Get a list of items for a task
+ * Get a paginated list of items for a task
  * @name  get-task-items
  * @param {Object} params - what the keys in the url mean
  * @param {String} params.task - the task id
  * @param {Object} query - what queryparams are valid
  * @param {String} [query.lock] - if provided this must either be 'unlocked' or 'locked'
+ * @param {String} [query.page=0] - starting page
+ * @param {String} [query.page_size=constants.PAGE_SIZE] - the size of the page
```
Comment (mcwhittemore) [inline lib/server.js:161]: Can you run the api docs script to see how this works?

File: lib/helper/sanitizeNumber.js (hunk 1 above)
Comment (kepta) [reply lib/helper/sanitizeNumber.js]: Ya, I thought about it. When testing negative number, Currently the server sends an error.
What are your thoughts on sending an error or correcting it for the user. ?

**Review by mcwhittemore — COMMENTED** [review]
@kepta this looks great. I left two small bits of feedback here.

File: lib/helper/sanitizeNumber.js (hunk 1 above)
Comment (mcwhittemore) [reply lib/helper/sanitizeNumber.js]: yea, that makes sense. I'm guessing that a 500 error which we'll want to write alarms against so anyplace where we can stop them from happening is best. Here we should return a 400 error as its bad use input.

**Comment by kepta** [comment]
@mcwhittemore I have fixed the things you pointed out. I would like to know your opinion on throwing an error inside the function and then catching it inside the express endpoint. 

Feel free to flag if I missed something. 

Merging this for now.

**Pull request merged**